proto/redpanda/api/dataplane/v1alpha1/pipeline.proto

```diff
@@ -0,0 +1,212 @@
+syntax = "proto3";
+
+package redpanda.api.dataplane.v1alpha1;
+
+import "buf/validate/validate.proto";
+import "google/api/field_behavior.proto";
+import "google/protobuf/field_mask.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+message Pipeline {
+  string id = 1 [
```

Comment:
Can we restrict the character set here as well? Ideally to something that's natively supported by SpiceDB as well. Similar naming constraints as we have for Kafka topic names would make sense to me.

Reply:
Done

Reply:
Should this be `IMMUTABLE` and `OUTPUT_ONLY` ?